lib/output.js

```diff
@@ -1,19 +1,119 @@
 "use strict";
 
-var fs = require("fs"),
+var assert = require("assert"),
+    os = require("os"),
     path = require("path"),
-    url = require("url");
+    url = require("url"),
+    util = require("util");
 
-var debug = require("debug"),
-    mkdirp = require("mkdirp");
+var AWS = require("aws-sdk"),
+    debug = require("debug"),
+    fse = require("fs-extra"),
+    holdtime = require("holdtime"),
+    mkdirp = require("mkdirp"),
+    Promise = require("bluebird"),
+    s3UploadStream = require("s3-upload-stream"),
+    tmp = require("tmp"),
+    _ = require("underscore");
 
-var s3Output = require("./s3_output");
+var log = debug("swfr:upload");
 
-module.exports = function(outputUri, done, fn) {
+var ensureDir = Promise.promisify(fse.ensureDir);
+var copy = Promise.promisify(fse.copy);
+
+tmp.setGracefulCleanup();
+
+var getContentType = function(extension) {
```

Comment:
Maybe use [`mime`](https://www.npmjs.com/package/mime) and register additional types / extensions?